This patch is based on the upstream commit described below, adapted for use in
the Debian package by Peter Michael Green.

commit 59c63d373804fe9d00d92d2dd47174d4e1637333
Author: Christian Duerr <contact@christianduerr.com>
Date:   Sun Oct 8 05:29:57 2023 +0200

    Update dependencies
    
    This patch applies all breaking and non-breaking dependency updates
    and bumps MSRV to 1.70.0.

Index: alacritty-terminal/src/tty/unix.rs
===================================================================
--- alacritty-terminal.orig/src/tty/unix.rs
+++ alacritty-terminal/src/tty/unix.rs
@@ -4,7 +4,8 @@ use std::ffi::CStr;
 use std::fs::File;
 use std::io::{Error, ErrorKind, Result};
 use std::mem::MaybeUninit;
-use std::os::unix::io::{AsRawFd, FromRawFd, RawFd};
+use std::os::fd::OwnedFd;
+use std::os::unix::io::{AsRawFd, FromRawFd};
 use std::os::unix::process::CommandExt;
 use std::process::{Child, Command, Stdio};
 use std::{env, ptr};
@@ -30,7 +31,7 @@ macro_rules! die {
 }
 
 /// Get raw fds for master/slave ends of a new PTY.
-fn make_pty(size: winsize) -> Result<(RawFd, RawFd)> {
+fn make_pty(size: winsize) -> Result<(OwnedFd, OwnedFd)> {
     let mut window_size = size;
     window_size.ws_xpixel = 0;
     window_size.ws_ypixel = 0;
@@ -188,12 +189,14 @@ fn default_shell_command(shell: &str, us
 /// Create a new TTY and return a handle to interact with it.
 pub fn new(config: &PtyConfig, window_size: WindowSize, window_id: u64) -> Result<Pty> {
     let (master, slave) = make_pty(window_size.to_winsize())?;
+    let master_fd = master.as_raw_fd();
+    let slave_fd = slave.as_raw_fd();
 
     #[cfg(any(target_os = "linux", target_os = "macos"))]
-    if let Ok(mut termios) = termios::tcgetattr(master) {
+    if let Ok(mut termios) = termios::tcgetattr(&master) {
         // Set character encoding to UTF-8.
         termios.input_flags.set(InputFlags::IUTF8, true);
-        let _ = termios::tcsetattr(master, SetArg::TCSANOW, &termios);
+        let _ = termios::tcsetattr(&master, SetArg::TCSANOW, &termios);
     }
 
     let user = ShellUser::from_env()?;
@@ -210,9 +213,9 @@ pub fn new(config: &PtyConfig, window_si
     // Ownership of fd is transferred to the Stdio structs and will be closed by them at the end of
     // this scope. (It is not an issue that the fd is closed three times since File::drop ignores
     // error on libc::close.).
-    builder.stdin(unsafe { Stdio::from_raw_fd(slave) });
-    builder.stderr(unsafe { Stdio::from_raw_fd(slave) });
-    builder.stdout(unsafe { Stdio::from_raw_fd(slave) });
+    builder.stdin(unsafe { Stdio::from_raw_fd(slave_fd) });
+    builder.stderr(unsafe { Stdio::from_raw_fd(slave_fd) });
+    builder.stdout(unsafe { Stdio::from_raw_fd(slave_fd) });
 
     // Setup shell environment.
     let window_id = window_id.to_string();
@@ -231,11 +234,11 @@ pub fn new(config: &PtyConfig, window_si
                 return Err(Error::new(ErrorKind::Other, "Failed to set session id"));
             }
 
-            set_controlling_terminal(slave);
+            set_controlling_terminal(slave_fd);
 
             // No longer need slave/master fds.
-            libc::close(slave);
-            libc::close(master);
+            libc::close(slave_fd);
+            libc::close(master_fd);
 
             libc::signal(libc::SIGCHLD, libc::SIG_DFL);
             libc::signal(libc::SIGHUP, libc::SIG_DFL);
@@ -261,12 +264,12 @@ pub fn new(config: &PtyConfig, window_si
             unsafe {
                 // Maybe this should be done outside of this function so nonblocking
                 // isn't forced upon consumers. Although maybe it should be?
-                set_nonblocking(master);
+                set_nonblocking(master_fd);
             }
 
             let mut pty = Pty {
                 child,
-                file: unsafe { File::from_raw_fd(master) },
+                file: unsafe { File::from_raw_fd(master_fd) },
                 token: mio::Token::from(0),
                 signals,
                 signals_token: mio::Token::from(0),
Index: alacritty-terminal/Cargo.toml
===================================================================
--- alacritty-terminal.orig/Cargo.toml
+++ alacritty-terminal/Cargo.toml
@@ -78,7 +78,9 @@ default-features = false
 version = "1.0.0"
 
 [target."cfg(unix)".dependencies.nix]
-version = "0.26"
+version = "0.27"
+default-features = false
+features = ["term"]
 
 [target."cfg(unix)".dependencies.signal-hook]
 version = "0.3.10"
